pkgs/development/tools/godot/mono/make-deps.nix

```diff
@@ -0,0 +1,71 @@
+{ godot-mono
+, lib
+, nuget-to-nix
+, writeScriptBin
+, path
+, stdenv
+}:
+
+# This can be used to recreate deps.nix. This entails doing things that are not permitted
+# with a sandboxed Nix build, so instead of building the deps.nix directly, it builds a
+# script that can be run to generate it.  Thus, to use this derivation, simply do:
+#
+# $ nix run <path-to-nixpkgs>#godot-mono.make-deps [outdir]
+#
+# The work will all be done in the specified outdir, and the resulting deps.nix will be
+# placed in it at the end. If you don't provide an outdir, a temporary one will be created
+# in your tmpfs. The final line of output will specify the path to the resulting deps.nix.
+
+godot-mono.overrideAttrs (self: base: {
+  pname = "godot-mono-make-deps";
+
+  nativeBuildInputs = base.nativeBuildInputs ++ [ nuget-to-nix ];
+
+  outputs = [ "out" ];
+  phases = [ "installRunnerPhase" ];
+
+  installRunnerPhase = ''
+    mkdir -p "$out"
+    cp -R "$runner"/. "$out"
+  '';
+
+  runner = writeScriptBin self.pname ''
+    #!${stdenv.shell}
+
+    if [ "$1" ]; then
+      mkdir -p "$1"
+
+      if [ "$(ls -A "$1")" ]; then
+        echo "Directory \"$1\" is not empty!" 1>&2
+        exit 1
+      fi
+    fi
+
+    nix-shell ${path} -A godot-mono.make-deps --run 'outdir="'"$(readlink -f "$1")"'"; eval "$makeDepsPhase"'
+  '';
+
+  configureSteps = lib.remove "configureNugetStep" base.configureSteps;
+
+  makeDepsPhase = ''
+    outdir=''${outdir:-$(mktemp -d)}
+
+    pushd "$outdir"
+      unpackPhase
+
+      cd source
+
+      patchPhase
+      eval "$configurePhase"
+      eval "$buildPhase"
+
+      nuget-to-nix buildhome/.nuget/packages > "$outdir"/deps.nix
+    popd
```

Comment:
This is a horrible hacky way to run things. Please sort those things into the right phases.

Reply:
What things would need to be sorted into phases exactly?  This is a script that runs within a nix shell and executes phases explicitly.  This is done this way specifically because it cannot be done within a nix build.

Reply:
It shouldn't be done like this. Use the normal phases and disable the one or two which shouldn't be run or write everything that is required into the script.

Reply:
Also, the reason I wrote the script as something that runs inside a nix-shell is so that it can use the unpack, configure, and build phases from the derivation.  This avoids the need to have a bash script that duplicates all the details of these steps.  It basically means we get a bash script that says "run these phases of this derivation in a non-sandboxed context, so that the build phase can access the internet".